src/lib/certificate-manager.ts

```diff
@@ -148,11 +152,51 @@ export async function openCertificate() {
 	shell.showItemInFolder( CA_CERT_PATH );
 }
 
+/**
+ * Checks if the root CA certificate is already trusted by the system
+ * @returns A promise that resolves to true if the certificate is trusted, false otherwise
+ */
+export async function isRootCATrusted(): Promise< boolean > {
+	if ( ! fs.existsSync( CA_CERT_PATH ) ) {
+		return false;
+	}
+
+	const platform = process.platform;
+	if ( platform === 'win32' ) {
+		try {
+			const { stdout } = await execFilePromise( 'certutil', [ '-store', 'ROOT', CA_NAME ] );
+			return stdout.includes( CA_NAME );
+		} catch ( error ) {
+			return false;
+		}
+	} else if ( platform === 'darwin' ) {
+		try {
+			const { stdout } = await execFilePromise( 'security', [
```

Comment:
I had a certificate that I had already trusted, so I marked it as untrusted in Key Chain to test it. The UI didn't display the trust helpers. When I removed the certificate from the Key Chain completely, the UI worked fine.

This command seems to check if the certificate exists. Should we test if it is trusted using the `security verify-cert` command?

This flow could be common:
1. User clicks "Trust Certificate"
2. User opens certificate so it's added to Key Chain

At this point, certificate exists, but is not trusted, so UI text disappears.

Reply:
+1. This is key.

Reply:
Completely right, I don't know how I missed that in my tests :)

Reply:
It works like a charm - it changes state as soon as I update it in Keychain Access and Focus Studio. Thanks for fixing that!